include/deal.II/base/smartpointer.h

```diff
@@ -396,6 +423,36 @@ SmartPointer<T, P>::operator=(const SmartPointer<T, P> &tt)
 
 
 
+template <typename T, typename P>
+inline SmartPointer<T, P> &
+SmartPointer<T, P>::operator=(SmartPointer<T, P> &&tt) noexcept
+{
+  // if objects on the left and right hand side of the operator= are
+  // the same, then this is a no-op
+  if (&tt != this)
+    {
+      // Let us unsubscribe from the current object
+      if (pointed_to_object_is_alive && t != nullptr)
+        t->unsubscribe(&pointed_to_object_is_alive, id);
```

Comment:
Same thing here - would this work without the `pointed_to_object_is_alive` check?